### ISSUE TITLE
`Buffer`s aren't recognized in browser bundles.

### ISSUE THREAD

**Issue by Shaptic** [stellar/js-xdr#93]
**TL;DR:** The latest version of this library does not work appear to work in browser-based environments.

### Summary
When combining the [`js-stellar-base@bump-js-xdr`](https://github.com/stellar/js-stellar-base/commits/bump-js-xdr) branch which is pointed at [`js-xdr@buffer-fixups`](https://github.com/stellar/js-xdr/commits/buffer-fixups) branch (necessary to make the package npm+git-able, plus making debugging easier) and running the browser tests there:

```bash
yarn build:browser && yarn test:browser
```

you get an error trace:

```
Chrome Headless 104.0.5112.101 (Linux x86_64) muxed account abstraction works parses M-addresses FAILED
	TypeError: XDR Read Error: source not specified
	     at new XdrReader (js-stellar-base/dist/stellar-base.js:63463:15)
	     at UnsignedHyper.fromXDR (js-stellar-base/dist/stellar-base.js:63847:20)
	     at _decodeAddressFullyToMuxedAccount (js-stellar-base/dist/stellar-base.js:59203:61)
	     at decodeAddressToMuxedAccount (js-stellar-base/dist/stellar-base.js:59088:12)
	     at new fromAddress (js-stellar-base/dist/stellar-base.js:24757:146)
	     at Context.<anonymous> (/tmp/_karma_webpack_777483/commons.js:62:16)
```

(Note that Node tests work just fine :tada:.)

### Investigation
I haven't nailed down the root cause nor a solution yet, but I'm hypothesizing that this is related to `Buffer` being polyfilled in the browser version. The stack trace leads us to this piece of code:

https://github.com/stellar/js-xdr/blob/6d5a107a905a00ac64ffa3facf4077f483f9baf9/src/serialization/xdr-reader.js#L11-L18

So the `source` isn't recognized as being a `Buffer`. I added some code to that constructor to inspect the object prototypes:

```js
    console.log(source);
    let p = source.__proto__;
    while (p !== null) {
      console.log(p);
      p = p.__proto__;
    }
```

which shows that it is, in fact, a `Buffer`:

```txt
Firefox 111.0 (Linux x86_64) LOG: Buffer{0: 0, 1: 0, 2: 0, 3: 0, 4: 0, 5: 0, 6: 0, 7: 0}
Firefox 111.0 (Linux x86_64) LOG: Buffer{parent: undefined, <tons more stuff>}
Firefox 111.0 (Linux x86_64) LOG: Uint8Array{}
Firefox 111.0 (Linux x86_64) LOG: TypedArray{}
Firefox 111.0 (Linux x86_64) LOG: Object{}
```

but the `instanceof` check still fails. 

**Hack**: If I change the check to `!(source instanceof Uint8Array)`, tests pass. 

However, this probably isn't an ideal workaround, so a proper solution is still pending. Also "**Why** doesn't this work?" is a bigger question.

**Comment by Shaptic** [comment]
Lmao I discovered [`Buffer.isBuffer(obj)`](https://nodejs.org/api/buffer.html#static-method-bufferisbufferobj) immediately after submitting this :facepalm: 

It still begs the question of why the `instanceof` check fails, but this is a better workaround for now.
